### DISCUSSION TITLE
No EBUS Communication with Valiant aroTherm Split and VR70

### DISCUSSION THREAD

**Discussion by ghost** [john30/ebusd#710] [Q&A]
Hello, 

i need some help to integrate my ebus Adapter 3 in my aroTherm Split and VR70 environment. 
The Adapter is connected via Usb and the Jumper is only connected to Pi 1 and 2. The yellow Led is on and the green one is flashing. The Ebus is connected to the ebus port of the VR70. 
If i want to scan the bus with the command 

`ebusd --scanconfig --logfile=/var/log/ebusd.log --loglevel=debug -f -d /dev/ttyUSB0`

if i check the info of the ebusd 
with `ebusctl info` 

I get this Information:

`version: ebusd 22.4.p20221105`
`device: /dev/ttyUSB0`
`signal: no signal`
`reconnects: 0`
`masters: 1`
`messages: 11`
`conditional: 0`
`poll: 0`
`update: 4`
`address 31: master #8, ebusd`
`address 36: slave #8, ebusd`

After start this is in the log file

`2022-11-06 12:32:40.139 [main notice] ebusd 22.4.p20221105 started with auto scan on device /dev/ttyUSB0`
`2022-11-06 12:32:40.139 [main info] loading configuration files from http://cfg.ebusd.eu/`
`2022-11-06 12:32:40.183 [main info] reading templates /`
`2022-11-06 12:32:40.226 [main info] read templates in /`
`2022-11-06 12:32:40.226 [main info] reading file memory.csv`
`2022-11-06 12:32:40.272 [main info] successfully read file memory.csv`
`2022-11-06 12:32:40.273 [main info] reading file broadcast.csv`
`2022-11-06 12:32:40.316 [main info] successfully read file broadcast.csv`
`2022-11-06 12:32:40.316 [main info] read config files, got 11 messages`
`2022-11-06 12:32:40.318 [bus notice] bus started with own address 31/36`
`2022-11-06 12:32:40.318 [main info] registering data handlers`
`2022-11-06 12:32:40.318 [main info] registered data handlers`

I think i have forgot or misunderstand something. I also tested it with another computer on the end of the usb port or another port but nothing change this situation that the signal is not found. 
Maybe someone get me some helping information to integrate the adapter and get some information of my heating system.

**Comment by ghost** [comment]
I forgot some other information. Maybe it is helpful. 

Ich added the "enh:" suffix to the device and the response in the log file was this:

`2022-11-06 13:41:32.372 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 13:41:32.376 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 13:41:32.380 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 13:41:32.389 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 13:41:32.393 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 13:41:32.437 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 13:41:32.481 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 13:41:32.525 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 13:41:32.569 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 13:41:32.612 [bus error] device status: unexpected available enhanced following byte 1`

**Comment by ThW-1** [comment]
I have exactly the same problem with the eBUS adapter v3.1 on a Vaillant heat pump VWF 117/4:

`2022-11-06 19:04:28.723 [main notice] ebusd 22.4.v22.4 started with auto scan on enhanced device  #/dev/ttyUSB1`
`2022-11-06 19:04:28.724 [main info] loading configuration files from http://cfg.ebusd.eu/`
`2022-11-06 19:04:28.797 [main info] reading templates /`
`2022-11-06 19:04:28.860 [main info] read templates in /`
`2022-11-06 19:04:28.860 [main info] reading file memory.csv`
`2022-11-06 19:04:28.904 [main info] successfully read file memory.csv`
`2022-11-06 19:04:28.904 [main info] reading file broadcast.csv`
`2022-11-06 19:04:28.948 [main info] successfully read file broadcast.csv`
`2022-11-06 19:04:28.948 [main info] read config files, got 11 messages`
`2022-11-06 19:04:28.954 [bus notice] device status: resetting`
`2022-11-06 19:04:28.955 [bus notice] bus started with own address ff/04`
`2022-11-06 19:04:28.955 [main info] registering data handlers`
`2022-11-06 19:04:28.955 [main info] registered data handlers`
`2022-11-06 19:04:29.006 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 19:04:29.032 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 19:04:29.049 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 19:04:29.092 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 19:04:29.135 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 19:04:29.177 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 19:04:29.220 [bus error] device status: unexpected available enhanced following byte 1`
`2022-11-06 19:04:29.263 [bus error] device status: unexpected available enhanced following byte 1`

**Comment by m0deller** [comment]
I have a 3.1 adapter with WiFi. Working ok with an aerotherm + ASHP. I'm using an earlier ebusd build however. Do you have another way to check the USB connection?

**Reply by ghost** [reply]
I have no idea to test the usb connection in another way. The Solution with the wifi modul is also my desired solution. But firstly i wanted to test it with the usb connection and test the comunnication with the EBus. 
At the moment i haven't solder the wifi modul to the ebus adapter. 
Maybe I can do it tonight, then I can test the wifi module to connect it.

**Reply by ThW-1** [reply]
Thanks for the reply, I am reassured to hear that the adapter version 3.1 works. Currently the adapter is connected to a Raspberry ARMv7 Processor rev 3 (bullseye). According to dmesg: usbserial: USB Serial support registered for cp210x / Product: FT230X Basic UART. I will connect a PC with linux for testing in the next days and then report.

**Comment by john30** [comment]
for adapter 3.1 the prefix needs to be "ens:" as stated in the [documentation](https://adapter.ebusd.eu/index.en.html#usb)

**Reply by ThW-1** [reply]
It works with "ens:". Thank you very much!

**Reply by ghost** [reply]
For me, the addition in the device of the command  

`ebusd --scanconfig --logfile=/var/log/ebusd.log --loglevel=debug -f -d ens:/dev/ttyUSB0`

also achieves the desired effect. Sometimes you can't see the forest full of trees. In any case, the problem is solved and I can start the integration into my environment. Many thanks for the support.
